content/platforms/kubernetes/managing-images.md

```diff
@@ -0,0 +1,295 @@
+namespace of interest on your cluster.
+
+Any image that is not prefixed by a repository domain name (e.g., "gcr.io"), will pull
+from the default repository for the Kubernetes cluster (i.e., DockerHub).
+
+For the Redis Enterprise Operator, you can determine your pull source by a
+simple filter:
+
+```
+kubectl get pods --all-namespaces -o jsonpath="{..image}" |tr -s '[[:space:]]' '\n' | uniq -c | grep redislabs
+```
+
+## Rating limiting with DockerHub
+
+Docker [announced restrictions](https://www.docker.com/blog/scaling-docker-to-serve-millions-more-developers-network-egress/)
+via rate limiting for the number of pulls, anonymous or authenticated,
+on the scale of 100 and 200 pulls per 6 hours, respectively. In the context of
+a Kubernetes cluster, this may affect the cluster in a number of ways:
+
+ * the cluster nodes will likely be treated as a one anonymous user,
+ * the number of pulls during an initial or subsequent deployment could
+   exceed the rate limiting for any aspect of the deployment: for our operator, Redis Enterprise, or for other non-Redis pods,
+ * pull failures may cause serious delays in the ability for a deployment to pull the necessary images and so will
+   affect the stability of deployments like the StatefulSet used by the Redis Enterprise operator.
+
+For this reason, serious consideration should be applied to where the images
+are pulled from to avoid failures caused by rate limiting. The easiest solution
+is to push the required images to a private repository under your control.
+
+## Managing image sources
+
+Redis Enterprise operator and cluster images are pushed to DockerHub,
+Red Hat, and other public registries for distribution. Your organization may
+require these images be copied to other registries used by your Kubernetes
+clusters.
+
+### Creating a private repository
+
+A private registry can be setup via:
+
+ * on-premise via [Docker registry](https://docs.docker.com/registry/deploying/),
+   RedHat Quay, or other providers,
+ * via a cloud provider (e.g., Azure Container Registry, Google Container Registry, etc.).
+
+Once you have setup a repository, you will have:
+
+ * a domain name,
+ * an optional port,
+ * an optional registry path suffix,
+
+This information identifies your registry and must be used to references images you
+push to your private registry. For example, a Google Container Registry
+will start with `gcr.io/{project-id}` where '{project-id}' is the cloud project
+identifier.
+
+### Pushing Redis Enterprise images to a private repository
+
+A Kubernetes deployment uses different images for various purposes:
+
+ * Redis Enterprise
+ * bootstrapping a Redis Enterprise cluster node
+ * the Service Rigger
+ * the Redis Enterprise operator
+
+Once you have setup a private repository, you will need to push these images
+to your private repository. The process to do so is:
+
+ 1. Pull the various images locally for Redis Enterprise and the operator.
```

Comment:
Various - because there can be more than one. Also, because it depends on which images you are pull and pushing and for what purpose. This is a general procedure.